lib/shopify_transporter/shopify/customer.rb

```diff
@@ -57,12 +57,36 @@ def to_csv
       ).freeze
 
       def top_level_row_values
-        base_hash.merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES)).values
+        return top_level_attributes.values if not_able_to_merge_address?
+
+        top_level_attributes.merge(default_address.compact).values_at(*Customer.columns)
+      end
+
+      def top_level_attributes
+        base_hash.merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES))
+      end
+
+      def default_address
+        return {} unless at_least_one_address?
+
+        record_hash['addresses'][0].slice(*ADDRESS_ATTRIBUTES)
+      end
+
+      def at_least_one_address?
+        record_hash['addresses'].present? && record_hash['addresses'][0].present?
+      end
+
+      def not_able_to_merge_address?
```

Comment:
functions like this should be written in the positive case, like `able_to_merge_address`, whenever possible. `not_` is a smell that you should be using logic operators in the code, like `!`, instead.

Reply:
Fixed in the latest commit